docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,254 @@
+        metav1.TypeMeta
+        Spec LLMBackendSpec
+}
+
+type LLMBackendPoolSpec struct {
+        // Pod selector, similar to k8s Service.
+        Selector map[string]string `json:"selector,omitempty"`
+        // Allows names within the `model` param that don't have an explicitly defined UseCase to pass through. Defaults to `false`.
+        AllowUndefinedModels bool
+        // Admin-defined minimum objective value that can be requested, will cause an error in LLMUseCase if limit is broken.
+        MinTPOT float32
+}
+```
+
+
+**LLMUseCase**
+```golang
+// LLMUseCase represents a use case of an LLM, which is multiplexed onto one or more LLMBackendPools.
+// It allows mapping a use case to backend pools, and models or adapters that backend model servers understand.
+// The LLMUseCase defines request routing behavior within an LLMBackendPool.
+// Generally the LLMUseCase is owned by the "LLM Use Case Owner" persona, which can be teams in an organization.
+type LLMUseCase struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+
+        Spec LLMUseCaseSpec
+}
+
+type LLMUseCaseSpec struct {
+        // Map use case to one or more backend pools.
+        // In the most common case, this should be a single backend pool.
+        // Multiple backend pools can be used for traffic splitting while migrating to a new backend pool.
+        Rules   []LLMUseCaseRule
+}
+
+// LLMUseCaseRule represents a mapping from a LLMUseCase to a backend pool and adapters/models in that pool.
+type LLMUseCaseRule struct {
+        // The name used in the `model` param of incoming requests
+        ModelName string
+        // Optional
+        Objective *Objective
+        // Required.
+        // Reference to an LLMBackendPool. This allows registering a use case
+        // as valid on a pool. 
+        // NOTE: Allowing multiple pools is a configuration convenience.
+        PoolRef []corev1.ObjectReference
+        // Optional.
+	    // Allow multiple versions of a model for traffic splitting. 
+	    // If not specified, the target model name is defaulted to the 
+        // modelName parameter.
+        TargetModels []common.TargetModel
+}
+
+
+// TargetModel represents a deployed model or a LoRA adapter.
+type TargetModel struct {
+        // The name of the adapter expected by the ModelServer.
+        TargetModelName string
+        // Weight is used to determine the percentage of traffic that should be 
+        // sent to this target model when
+        // multiple versions of the models are specified.
+        Weight int
+}
+
+// Objective defines the performance targets of a LLM use case.
+// NOTE: Objectives are best effort
+type Objective struct {
+        // Only one target can be set.
+        TPOT    []LatencyTarget
+        FairnessWeight int
+}
+
+
+type LatencyTarget struct {
+        Percentile float64       `json:"percentile,omitempty"`
+        Target       time.Duration `json:"target,omitempty"`
+}
+```
+
+### Diagrams
```

Comment:
I think it is useful to add a CUJs section to show how the API can be used to represent them.

Reply:
Ack, adding